core/lib/eth_client/src/clients/http/query.rs

```diff
@@ -22,29 +26,44 @@ use crate::{
 
 /// An "anonymous" Ethereum client that can invoke read-only methods that aren't
 /// tied to a particular account.
-#[derive(Debug, Clone)]
+#[derive(Clone)]
 pub struct QueryClient {
-    web3: Arc<Web3<Http>>,
+    web3: Web3<Http>,
+    url: SensitiveUrl,
 }
 
-impl From<Http> for QueryClient {
-    fn from(transport: Http) -> Self {
-        Self {
-            web3: Arc::new(Web3::new(transport)),
-        }
+impl fmt::Debug for QueryClient {
+    fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        formatter
+            .debug_struct("QueryClient")
+            .field("url", &self.url)
+            .finish()
     }
 }
 
 impl QueryClient {
     /// Creates a new HTTP client.
-    pub fn new(node_url: &str) -> Result<Self, Error> {
-        let transport = Http::new(node_url)?;
-        Ok(transport.into())
+    pub fn new(url: SensitiveUrl) -> Result<Self, Error> {
+        let transport = Http::new(url.expose_str())?;
+        Ok(Self {
+            web3: Web3::new(transport),
+            url,
+        })
     }
 }
 
 #[async_trait]
 impl EthInterface for QueryClient {
+    async fn fetch_chain_id(&self, component: &'static str) -> Result<L1ChainId, Error> {
```

Comment:
I have a follow-up PR refactoring L1 clients almost prepared, but it depends on the code in this PR, so I'll publish it after this one is merged.